firmware/shared/util/algorithms/arrays.h

```diff
@@ -21,14 +22,14 @@ concept Numeric = requires { std::is_arithmetic_v<T>; };
 /// of the minimum value, the least index is returned.
 /// @return Minimum value of the array.
 /// @note Returns a large value if `array_length = 0`.
-template <Numeric T, int array_length>
-    requires(array_length >= 0)
-T GetMinimum(T* array, int* idx_min) {
+template <Numeric T, int array_length, Numeric I>
```

Comment:
does the additional template parameter provide any benefit besides checking limits? We really should be using a `size_t` variable for the index return, then the array length limit is unnecessary. It is the responsibility of the programmer to convert the `size_t` value into whatever type they need.

It seems like you added this to help get the `uint8_t high_thermistor_idx` value for TMS. I don't think we should be performing this check for the programmer; they should be aware of the inherent limits of their chosen datatype.

Reply:
why not have the check built in when its easily available? I quite like when we are able to remove the possibility of a mistake with a compile time check.